golang-ci/example-project/main_test.go

```diff
@@ -0,0 +1,56 @@
+package main
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+)
+
+func TestHelloHandler(t *testing.T) {
+	req, err := http.NewRequest("GET", "/hello", nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	rr := httptest.NewRecorder()
+	handler := http.HandlerFunc(helloHandler)
+
+	handler.ServeHTTP(rr, req)
+
+	if status := rr.Code; status != http.StatusOK {
+		t.Errorf("handler returned wrong status code: got %v want %v",
+			status, http.StatusOK)
+	}
+}
+
+func TestHelloHandler404(t *testing.T) {
```

Comment:
🤔  is this too much code? I think actually all we need is a single test to ensure the test step does what we think it does.

There's nothing specific to this pipeline that actually requires a web server running so we could almost drop all assertions and replace it with the following:

```go
func TestSomething(t *testing.T) {}
```